packages/language-server/src/server.ts

```diff
@@ -155,6 +153,12 @@ connection.onInitialize(async (params) => {
 })
 
 connection.onInitialized(async () => {
+	if (capabilities.textDocument?.formatting?.dynamicRegistration) {
+		void connection.client.register(
+			ls.DocumentFormattingRequest.type,
+			{ documentSelector: [{ language: 'mcdoc' }] },
```

Comment:
What's the main purpose of doing this as a dynamic registration? Is it not possible to configure the `documentFormattingProvider` above and give it a document selector?